core/src/main/java/io/lionweb/lioncore/java/serialization/PrimitiveValuesSerialization.java

```diff
@@ -152,11 +162,20 @@ public String serialize(@Nonnull String primitiveTypeID, @Nullable Object value)
         return null;
       }
       // In this case, where we are dealing with primitive values, we want to use the literal _key_
-      // (and not the ID)
+      // (and not the ID).
       // This is at least the default behavior, but the user can register specialized
-      // primitiveSerializers,
-      // if a different behavior is needed
-      return ((EnumerationLiteral) value).getKey();
+      // primitiveSerializers, if a different behavior is needed
+      EnumerationLiteral enumerationLiteral;
+      if (value instanceof EnumerationValue) {
+        enumerationLiteral = ((EnumerationValue) value).getEnumerationLiteral();
```

Comment:
Why not `return ((EnumerationValue) value).getEnumerationLiteral().getKey();` here, then? The `else`-branch just `throw`s.